opendevin/runtime/client/client.py

```diff
@@ -60,26 +61,51 @@ def __init__(self, plugins_to_load: list[Plugin], work_dir: str) -> None:
 
     def _init_bash_shell(self, work_dir: str) -> None:
         self.shell = pexpect.spawn('/bin/bash', encoding='utf-8', echo=False)
-        self.__bash_expect = r'\[PEXPECT\][\$\#] '
-        self.__bash_PS1 = r'\u@\h:\w [PEXPECT]\$ '
+        self.__bash_PS1 = r'[PEXPECT_BEGIN] \u@\h:\w [PEXPECT_END]'
+
+        # This should NOT match "PS1=\u@\h:\w [PEXPECT]$" when `env` is executed
+        self.__bash_expect_regex = r'\[PEXPECT_BEGIN\] ([a-z_][a-z0-9_-]*)@([a-zA-Z][a-zA-Z0-9.-]*):(.+) \[PEXPECT_END\]'
+
         self.shell.sendline(f'export PS1="{self.__bash_PS1}"')
-        self.shell.expect(self.__bash_expect)
+        self.shell.expect(self.__bash_expect_regex)
+
         self.shell.sendline(f'cd {work_dir}')
-        self.shell.expect(self.__bash_expect)
+        self.shell.expect(self.__bash_expect_regex)
+
+    def _get_bash_prompt(self):
+        ps1 = self.shell.after
+        # parse the ps1 to get username, hostname, and working directory
+        matched = re.match(self.__bash_expect_regex, ps1)
+        assert (
+            matched is not None
+        ), f'Failed to parse bash prompt: {ps1}. This should not happen.'
+        username, hostname, working_dir = matched.groups()
+
+        # re-assemble the prompt
+        prompt = f'{username}@{hostname}:{working_dir} '
+        if username == 'root':
+            prompt += '#'
+        else:
+            prompt += '$'
```

Comment:
I didn't try locally. Does different `username` will show different prompts in terminal?

Reply:
Yep - this seems like an interesting thing in bash:
https://stackoverflow.com/questions/18272722/why-does-my-root-user-have-a-different-shell-prompt-to-other-users-with-the-same